### ISSUE TITLE
AudioEffectDistortion in an Audio Bus that is playing, affects other Audio Bus sound

### ISSUE THREAD

**Issue by oscarduretto** [godotengine/godot#38372]
**Godot version:** v3.2.1.stable.official

**OS/device including version:** Windows 10

**Issue description:** When an AudioEffectDistortion is added to a non-Master bus, and that Audio bus has music playing within it, then sounds being played from other buses are distorted. This even occurs if the bus with the Distortion effect on it is Muted or sound is lowered so you can not hear it. The Distortion on the other buses sounds different and seems more distorted than the original Distortion bus (possibly being corrupted in some other manner)

I assume this is not expected behaviour as AudioEffects should only apply to the music/sound that is playing through their Audio Bus and have no effect on other Audio Buses. Annoying if you wish to apply distortion to a single song or sound effect while not affecting other sounds playing.

**Steps to reproduce:**

This is replicated in the editor. Though I discovered the issue through a coded setup (adding effects and playing music through GDScript).

- Create two audio buses (bus names: 'distorted', 'non-distorted' for clarity).
- Add a Distortion effect (no modification) to 'distorted' bus.
- Create a scene with two AudioStreamPlayers. One using the 'distorted' bus, one using the 'non-distorted' bus and provide them with looping music (.ogg file, untested with other types of resource). Using the same .ogg resource or different does not change anything.
- Mute or reduce the 'distorted' bus db so you can not hear it when it plays.
- Set both AudioStreamPlayers to playing.
- You will hear a distorted version of the song that is playing on the 'non-distorted' bus AudioStreamPlayer. If you turn off the Distorted effect on the 'distorted' bus the effect will stop. Also if you stop the 'distorted' bus AudioStreamPlayer from playing its music, the effect will stop. This distortion also sounds more distorted if you compare it to the normal Distortion effect.

- This does not replicate for any other effects attempted (tried Reverb, Phaser and Delay). I did not thoroughly test all effects.

**Minimal reproduction project:**

The projects contains a song by Kevin Mcleod I was using for testing. I was able to replicate the results with other music and .ogg files.
Onion Capers by Kevin MacLeod
Link: https://incompetech.filmmusic.io/song/4679-onion-capers
License: http://creativecommons.org/licenses/by/4.0/

[test-audio.zip](https://github.com/godotengine/godot/files/4562325/test-audio.zip)

**Comment by tom-jk** [comment]
Can reproduce in 3.2.2.stable on linux mint 18.3 with supplied project (though I had to swap out the given music for another ogg as project was crashing on load).

**Comment by ellenhp** [comment]
Current theory: The distortion effect is intermittently putting a bunch of NaNs in the audio buffer. It might be caused by the extremely quiet input, because the effect sounds fine normally. The NaNs propagate all the way through the mixing and cause discontinuities in the output, which is why it sounds more distorted than the normal distortion effect.

All I've confirmed of this theory though is that there **definitely** NaNs in the output of the distortion filter. I haven't really investigated the cause or solution yet. It's late here, but after jumping the gun thinking I had a fix earlier I didn't feel like I could sleep until I figured this one out. Now I can sleep. :)
